docs/account_system.md

```diff
@@ -0,0 +1,158 @@
+    "password": "<password>",
+    "realName": "<Real Name>",
+    "captchaResponse": "<reCaptcha response data>"
+}
+```
+
+Response:
+```JSON
+{
+	"success": boolean,
+	"errorMessage": "<error message>",
+	"sessionKey": "<base64 session key>",
+}
+```
+
+### `/user/login`
+
+Request a new session. This should only be done if the current session key is invalid or missing (on a new device or when cookies were cleared for instance).
+For security reasons, only 5 simultaneous sessions are allowed; the oldest session will automatically be invalidated if a client goes over the limit.
+
+Query:
+```JSON
+{
+    "username": "<username>",
+    "password": "<password>",
+    "captchaResponse": "<reCaptcha response data>"
+}
+```
+`captchaResponse` is only needed when the user or client is rate-limited (too many login attempts in the last hour).
+
+Response:
+```JSON
+{
+	"success": boolean,
+    "rateLimited" boolean,
+	"errorMessage": "<error message>",
+	"sessionKey": "<base64 session key>"
+}
+```
+If `rateLimited` is `true`, retry login with CAPTCHA.
+
+### `/user/info`
+
+Request information about the (presumably) logged in user. This should be used every time a page is loaded.
+
+Query:
+```JSON
+{
+    "sessionKey": "<session key>"
+}
+```
+
+Response:
+```JSON
+{
+    "loggedIn": boolean,
+    "username": "<username>",
+    "realName": "<Real Name>"
+}
+```
+If the user is not logged in (the sessionKey is invalid or has been disabled, see "Expected Client Behavior"), userName and realName will be blank.
+
+### `/user/update/password`
+
+Change the logged in user's password. Password strength checking is done client-side.
+
+Query:
+```JSON
+{
+    "sessionKey": "<session key>",
+    "password": "<password>"
+}
+```
+
+Response:
+```JSON
+{
+	"success": boolean,
+	"errorMessage": "<error message>"
+}
+```
+
+### `/user/update/realName`
+
+Change the logged in user's real name. Empty field is allowed.
+
+Query:
+```JSON
+{
+    "sessionKey": "<session key>",
+    "realName": "<Real name>"
+}
+```
+
+Response:
+```JSON
+{
+	"success": boolean,
+	"errorMessage": "<error message>"
+}
+```
+
+SQL Structure
+-------------
+
+```SQL
+create table users (
+    id integer not null primary key,
+    username text not null,
+    passwordHash text not null,
+    realName text not null,
+    disabled integer not null
+);
+
+create table userSessions (
+    sessionKey text not null primary key,
+    userId integer not null,
+    loginTime integer not null,
+    lastSeenTime integer not null
+);
+
+create table userLogins (
+    id integer not null primary key,
+    userId integer not null,
+    remoteIp text not null,
+    time integer not null,
+    success integer not null
+);
+```
+
+* `users` contains the user account information;
+* `userSessions` contains all active session (*however, an active session is not necessarily a valid session; if it is invalid/too old/whatever, a login check must ask for re-authentication and delete the session*);
```

Comment:
The OAuth way of doing this is a 'refresh token': tokens have a finite life (even if you keep using them), and eventually have to be refreshed using the 'refresh token' credential.